python/mlx/optimizers.py

```diff
@@ -221,3 +221,44 @@ def apply_single(
         return super().apply_single(
             gradient, parameter * (1 - self.learning_rate * self.weight_decay), state
         )
+
+
+class Adagrad(Optimizer):
+    r"""Implementation of the Adagrad optimizer [1].
+
+    Our Adagrad implementation follows the original paper and omits the bias
+    correction in the first and second moment estimates. In detail,
```

Comment:
I think this has been left from copying from Adam as there are no moment estimates.